homeassistant/components/hive/__init__.py

```diff
@@ -73,9 +137,59 @@ def setup(hass, config):
     session.attributes = Pyhiveapi.Attributes()
     hass.data[DATA_HIVE] = session
 
-    for ha_type, hive_type in DEVICETYPES.items():
-        for key, devices in devicelist.items():
-            if key == hive_type:
-                for hivedevice in devices:
-                    load_platform(hass, ha_type, DOMAIN, hivedevice, config)
+    for ha_type in DEVICETYPES:
+        devicelist = devices.get(DEVICETYPES[ha_type])
+        if devicelist:
+            load_platform(hass, ha_type, DOMAIN, devicelist, config)
+            if ha_type == "climate":
+                hass.services.register(
+                    DOMAIN,
+                    SERVICE_BOOST_HEATING,
+                    heating_boost,
+                    schema=BOOST_HEATING_SCHEMA,
+                )
+            if ha_type == "water_heater":
+                hass.services.register(
+                    DOMAIN,
+                    SERVICE_BOOST_HEATING,
+                    hotwater_boost,
+                    schema=BOOST_HOTWATER_SCHEMA,
+                )
+
     return True
+
+
+def refresh_system(func):
+    """Force update all entities after state change."""
+
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        func(self, *args, **kwargs)
+        dispatcher_send(self.hass, DOMAIN)
+
+    return wrapper
+
+
+class HiveEntity(Entity):
+    """Initiate Hive Base Class."""
+
+    def __init__(self, session, hive_device):
+        """Initialize the instance."""
+        self.node_id = hive_device["Hive_NodeID"]
+        self.node_name = hive_device["Hive_NodeName"]
+        self.device_type = hive_device["HA_DeviceType"]
+        self.node_device_type = hive_device["Hive_DeviceType"]
```

Comment:
This attribute seems only used in the binary sensor platform. We can move it to that class.

Reply:
@MartinHjelmare , we do populate that data for all the other devices/platforms. Is it worth leaving it in the parent class in case that data is used in the future with other platforms other than binary_sensor?

Reply:
Ok, let's keep it.